nucliadb_one/nucliadb_one/tests/test_fieldmetadata.py

```diff
@@ -0,0 +1,153 @@
+        ],
+        "token": [
+            {"token": "token2", "klass": "klassB", "start": 5, "end": 12},
+            {"token": "token3", "klass": "klassC", "start": 15, "end": 16},
+        ],
+    }
+
+    # Step 1
+
+    async with nucliadb_api(roles=[NucliaDBRoles.WRITER]) as client:
+        resp = await client.post(
+            f"/{KB_PREFIX}/{knowledgebox_one}/resources",
+            headers={"X-SYNCHRONOUS": "True"},
+            json={
+                "texts": {
+                    "textfield1": {"body": "Some text", "format": "PLAIN"},
+                    "textfield2": {"body": "Some other text", "format": "PLAIN"},
+                },
+                "fieldmetadata": [fieldmetadata_0],
+            },
+        )
+        assert resp.status_code == 201
+        rid = resp.json()["uuid"]
+
+    async with nucliadb_api(roles=[NucliaDBRoles.READER]) as client:
+        resp = await client.get(f"/{KB_PREFIX}/{knowledgebox_one}/resource/{rid}")
+        fieldmetadata = resp.json()["fieldmetadata"]
+        assert len(fieldmetadata) == 1
+        assert fieldmetadata[0] == fieldmetadata_0
+
+    # Step 2
+
+    async with nucliadb_api(roles=[NucliaDBRoles.WRITER]) as client:
+        resp = await client.patch(
+            f"/{KB_PREFIX}/{knowledgebox_one}/resource/{rid}",
+            headers={"X-SYNCHRONOUS": "True"},
+            json={"fieldmetadata": [fieldmetadata_1]},
+        )
+        assert resp.status_code == 200
+
+    async with nucliadb_api(roles=[NucliaDBRoles.READER]) as client:
+        resp = await client.get(
+            f"/{KB_PREFIX}/{knowledgebox_one}/resource/{rid}?show=basic&show=extracted",
+        )
+        assert resp.status_code == 200
+        fieldmetadata = resp.json()["fieldmetadata"]
+
+        assert len(fieldmetadata) == 2
+        assert fieldmetadata[0] == fieldmetadata_0
+        assert fieldmetadata[1]["field"] == fieldmetadata_1["field"]
+        assert fieldmetadata[1]["paragraphs"] == fieldmetadata_1["paragraphs"]
+        assert fieldmetadata[1]["token"] == []
+
+    # Step 3
+
+    async with nucliadb_api(roles=[NucliaDBRoles.WRITER]) as client:
+        resp = await client.patch(
+            f"/{KB_PREFIX}/{knowledgebox_one}/resource/{rid}",
+            headers={"X-SYNCHRONOUS": "True"},
+            json={"fieldmetadata": [fieldmetadata_2]},
+        )
+        assert resp.status_code == 200
+
+    async with nucliadb_api(roles=[NucliaDBRoles.READER]) as client:
+        resp = await client.get(
+            f"/{KB_PREFIX}/{knowledgebox_one}/resource/{rid}?show=basic&show=extracted",
+        )
+        assert resp.status_code == 200
+        fieldmetadata = resp.json()["fieldmetadata"]
+
+        assert len(fieldmetadata) == 2
+        assert fieldmetadata[0] == fieldmetadata_2
+        assert fieldmetadata[1]["field"] == fieldmetadata_1["field"]
+        assert fieldmetadata[1]["paragraphs"] == fieldmetadata_1["paragraphs"]
+        assert fieldmetadata[1]["token"] == []
```

Comment:
What about adding some dummy tokens on the test too?
This way we make sure that the `start` and `end` fields that we added are covered by a test.

Reply:
`start` and `end` are implicitly tested on the asserts like `assert fieldmetadata[0] == fieldmetadata_2`. But I can add a more explicit one if you think so